src/components/Catalog.js

```diff
@@ -0,0 +1,48 @@
+import { Component } from "react";
+import Movie from './Movie';
+import Rented from './Rented'
+
+class Catalog extends Component {
+
+    constructor() {
+        super();
+        this.state = {
+            search: '',
+            budget: 10
+        } 
+    }
+
+    updateSearch = (e) => {
+        this.setState({search: e.target.value}) 
+    }
+
+    manngeAdding = (movieID) => {
+        if (this.state.budget < 3){
+            alert("You don't have enough money for renting another movie");
+            return;
+        }
+        this.props.addMovie(movieID);
+        this.setState({budget: this.state.budget - 3})
+    }
+
+    mannageRemoving = (movieID) =>{
+        this.props.removeMovie(movieID);
+        this.setState({budget: this.state.budget + 3})
+    }
+
+    render() {
+        return (
```

Comment:
you use this.state.search a lot. If you are using something a lot, just redefine it so it is shorter in the rest of your code
`const {search, budget} = this.state`